packages/components/src/vue/components/panel-compact.vue

```diff
@@ -493,7 +493,12 @@ export default {
           if (lexeme.meaning.shortDefs.length > 0) {
             definitions.push(...lexeme.meaning.shortDefs)
           } else if (Object.entries(lexeme.lemma.features).length > 0) {
-            definitions.push(new Definition(this.l10n.getMsg('TEXT_NOTICE_NO_DEFS_FOUND'), 'en-US', 'text/plain', lexeme.lemma.word))
+            definitions.push(
+              new Definition(
+                this.l10n.getMsg('TEXT_NOTICE_NO_DEFS_FOUND'), Language.ENGLISH,
+                Constants.MIMETypes.TEXT_PLAIN, lexeme.lemma.word
```

Comment:
Yes, those two are different, but I believe I was fixing a mistake that we made before. Here we create an instance of a `Definition` class, and the second parameter of its constructor is the language on what the text of the definition is written.

Previously, the `Definition` constructor was taking a language code. I've inspected the usages and in absolute majority of cases it was `eng` with `en` being the rest. `en-US` is the locale. Locale, to the best of my understanding, expresses user preferences related to the location, such as date, time formats, and may, or may not, depending on the implementation, control the language on which messages are written (for example, in MS Visual Studio locale and language are two different concepts: https://github.com/alpheios-project/alpheios-core/issues/597). 

The `Definition`, especially since its parameter was switched to be the `Language` object, is clearly requiring a language, not a locale. So the `Language.ENGLISH` (with the language code `eng`), not the `en-US` locale, is the most appropriate argument, on my opinion.

This, however, rises an issue I didn't realize before: in cases like that we need not to hardcode the language, but take it from the `l10n` instance (what is the language of the message that was just returned?). So maybe we should modify the `L10n` class to return the language of the currently selected console. 

@balmas, @irina060981: What do you think?

Reply:
This is definitely a confusing area, because there are 2 different language values that are pertinent to a definition:
1) the language of the lemma which the definition is a definition of
2) the language of the text of the definition

Currently, our code has the first (the language of the lemma) as a property of the DefinitionSet that contains the Definition and the second (the language of the text of the definition) as a property of the Definition itself.

The language of the text of the definition is set by whatever source that the definition comes from, and in theory could be either an ISO-2 or ISO-3 code.  It may or may not be a language that we have a Language Model class for (i.e. one that Alpheios supports as a source text.)

The question here is related to others raised in this issue - whether the Language object should be used to encapsulate all concepts of Language or just those that we have a Language model for.  I'm not sure of the answer.

Reply:
According to the current case, I believe, we have several different issues:

We have a language of the targetWord and we have a language of the Definition (_1 issue_) (that should be defined by definition's source and user's locale, but in our case we always have there the same `en-US`)
the language of definition would be used once inside definition vue component (example from short definition):
```
<span ... v-html="definition.text" :lang="definition.language"></span>
```

This Definitions's language is defined in three places:
1. inside lexicons adapter - from the local config file (each lexicon source has a predefined language of the definitions)
2. inside tufts adapter for some languages (lat, syr) - I believe from the remote source
3. inside `panel-compact.vue` (I think it is not a really good place for that) (_2 issue_) - to define an empty Definition (_3 issue_)

I like the following approach:

--Issue 1--
- use Language class as a class for all places where we use Language (for consistency)
- define several getters:
   - getter `code` (instead of toCode() - for better consistency) - it would return default getter `codeISO2` 
   - getter `codeISO2` - `Language.English => 'eng'`
   - getter `codeISO1US` - `Language.English => 'en-US'`
   - getter `codeISO1UK` - `Language.English => 'en-UK'`
- use Language class for Lemma, Definition, Translation, Word Usage and so on, where language inside each of the instance defines the language of saved text in it 

--Issue 2,3-- 
- define defaultLanguage, emptyText, defaultFormat
- define a static method inside Definition object - `createEmptyDefinitionObject(lemmaText)`
- use this static method inside panel-compact instead of direct definition 

Also I think it would be good to define more clearly the defnition's language inside Tufts adapter similiar to Lexicon's adapter

@balmas and @kirlat , what do you think?

Reply:
> The question here is related to others raised in this issue - whether the Language object should be used to encapsulate all concepts of Language or just those that we have a Language model for. I'm not sure of the answer.

I tend to think that the `Language` class should be an instance representing just the language code. We probably might not need to force its conversion (but maybe have an option to do that) to any particular ISO 639 format, but rather keep it as provided, as long as it conforms to either of ISO 639‑1, ISO 639‑2, and ISO 639‑3. But, as mappings between those formats are often possible (as https://iso639-3.sil.org/about/relationships states), we can utilize it during comparison, at least. We can also use specific getters to convert from one format to the other, as @irina060981 suggests. Probably it would make sense to use pre-existing libraries for that, such as https://github.com/wooorm/iso-639-3. I don't know if this one is the best, I did not have chance to research it in depth, but I think solutions must exist.

I'm thinking of the language model, on the other hand, as of something bigger than just a language. To me it's more like a set of language concepts that we define for languages we use. So, I think, we use `Language` when we need to store what is the language. We use `LanguageModel` when we need to refer to the extended linguistic information available for the language. That would not always be the case. If we have a definition in German, for example, we'll never need to use the language model for that Same is probably the case for other languages of definition texts: we don't expect to do any transformations that would require extended language data. But if we're working with the lexeme in Latin, for example, we'll need to pull in the Latin Language Model, because without information contained in that object we won't be able to do all the transformations of the lexeme that we need to do. Does it make sense?

Reply:
> use Language class as a class for all places where we use Language (for consistency) define several getters:
> * getter code (instead of toCode() - for better consistency) - it would return default getter codeISO2
> * getter codeISO2 - Language.English => 'eng'
> * getter codeISO1US - Language.English => 'en-US'
> * getter codeISO1UK - Language.English => 'en-UK'
> use Language class for Lemma, Definition, Translation, Word Usage and so on, where language inside each of the instance defines the language of saved text in it

I generally agree with this approach, I'm just not sure what getters should return exactly, and how should they be named, consequently. We might not need them at all. A comparison function that can compare two languages with different language codes might be enough. But we should analyze use cases carefully to get the right answer. I think we should try to minimize the amount of code we right and support in order to keep things maintainable.

Reply:
> * define defaultLanguage, emptyText, defaultFormat
> * define a static method inside Definition object - createEmptyDefinitionObject(lemmaText)
> * use this static method inside panel-compact instead of direct definition

I might be not fully familiar with all the use cases, but I'm against creating the `Definition` object without a language. If we think the language is a part of the minimal viable object state, and because it's used to calculate the content-based ID, we should always create a `Definition` object with the language defined and never change it afterwards. If it changes, then we need to create a new `Definition` instance instead. That is if we consider `Definition` to be a true value object, which I think it is.

In a situation of a compact panel use case, where we create a definition out of a localized message, we should ideally have a method of `L10n` that would return a language of the currently selected locale, and then try to create a message on that language. If it is not available, we should use the message on the default language. What do you think?